### PR TITLE
added cluster registration apis in prism-go-client and its unit tests

### PR DESCRIPTION
**Pull request by deepakm-ntnx**
- added cluster registration api client stubs 
- added its unit tests to validate apis

### CODE REVIEW

**Review by johnliu25 — DISMISSED** [review]

File: karbon/karbon_cluster_registration_service_test.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+)
+
+func printK8sClusterRegistration(k8sClusterReg *K8sClusterRegistration) {
+	fmt.Printf("Type: K8sClusterRegistration\n")
+	fmt.Printf("|-Name: %s\n", *k8sClusterReg.Name)
+	fmt.Printf("|-Status: %s\n", k8sClusterReg.Status)
+	fmt.Printf("|-UUID: %s\n", k8sClusterReg.UUID)
+	// fmt.Printf("|-K8s Identity Name: %s\n", k8sClusterReg.Identity.Name)
+	// fmt.Printf("|-K8s Identity UUID: %s\n", *k8sClusterReg.Identity.UUID)
+	// fmt.Printf("|-K8s Identity Kind: %s\n", *k8sClusterReg.Identity.Kind)
```
Comment (thunderboltsid) [inline karbon/karbon_cluster_registration_service_test.go:22]: please remove commented code in ht function.

Comment (deepakm-ntnx) [reply karbon/karbon_cluster_registration_service_test.go]: added asserts and comments with TODO

File: karbon/karbon_cluster_registration_service_test.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	interceptor := khttpclient.NewInterceptor(http.DefaultTransport)
+	creds := testCredsFromEnv(t)
+	nkeClient, err := NewKarbonAPIClient(creds, WithRoundTripper(interceptor))
+	require.NoError(t, err)
+
+	kctx := mock.NewContext(mock.Config{
+		Mode: keploy.MODE_TEST,
+		Name: t.Name(),
+	})
+	fmt.Println("Get Cluster registration List")
```
Comment (thunderboltsid) [inline karbon/karbon_cluster_registration_service_test.go:130]: Can we remove this or switch this to `t.Log` ?

File: karbon/karbon_cluster_registration_service_test.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+		fmt.Println("Get Cluster registration exists")
+		validateK8sClusterRegistrationGetResponse(t, test_cluster_name, test_cluster_uuid, responseGetReg)
+		// Registration exists. delete it so that we can create it
+		fmt.Println("Delete Cluster registration")
+		responseDelReg, err := nkeClient.ClusterRegistrationOperations.DeleteK8sRegistration(kctx, test_cluster_uuid)
+		assert.NoError(t, err)
+		validateK8sClusterRegistrationDeleteResponse(t, test_cluster_name, test_cluster_uuid, responseDelReg)
+	}
+
+	fmt.Println("Create Cluster registration")
```
Comment (thunderboltsid) [inline karbon/karbon_cluster_registration_service_test.go:104]: Can we remove this or switch this to `t.Log` ?

File: karbon/karbon_cluster_registration_service_test.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	test_metadata_apiversion := "v1.1.0"
+	test_metadata := &Metadata{APIVersion: &test_metadata_apiversion}
+
+	fmt.Println("Get Cluster registration")
+	responseGetReg, err := nkeClient.ClusterRegistrationOperations.GetK8sRegistration(kctx, test_cluster_uuid)
+	if err == nil {
+		fmt.Println("Get Cluster registration exists")
+		validateK8sClusterRegistrationGetResponse(t, test_cluster_name, test_cluster_uuid, responseGetReg)
+		// Registration exists. delete it so that we can create it
+		fmt.Println("Delete Cluster registration")
```
Comment (thunderboltsid) [inline karbon/karbon_cluster_registration_service_test.go:98]: Can we remove this or switch this to `t.Log` ?

File: karbon/karbon_cluster_registration_service_test.go [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+		"KubernetesClusterName": test_cluster_name,
+		"KubernetesClusterUUID": test_cluster_uuid,
+	}
+	test_metadata_apiversion := "v1.1.0"
+	test_metadata := &Metadata{APIVersion: &test_metadata_apiversion}
+
+	fmt.Println("Get Cluster registration")
+	responseGetReg, err := nkeClient.ClusterRegistrationOperations.GetK8sRegistration(kctx, test_cluster_uuid)
+	if err == nil {
+		fmt.Println("Get Cluster registration exists")
```
Comment (thunderboltsid) [inline karbon/karbon_cluster_registration_service_test.go:95]: Can we remove this or switch this to `t.Log` ?

File: karbon/karbon_cluster_registration_service_test.go [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+	test_cluster_uuid := strings.ToLower("634F5852-CE1B-465C-A95A-9B8DFFBDDE42")
+	test_category_mapping := map[string]string{
+		fmt.Sprintf("kubernetes-io-cluster-%s", test_cluster_name): "owned",
+		"KubernetesClusterName": test_cluster_name,
+		"KubernetesClusterUUID": test_cluster_uuid,
+	}
+	test_metadata_apiversion := "v1.1.0"
+	test_metadata := &Metadata{APIVersion: &test_metadata_apiversion}
+
+	fmt.Println("Get Cluster registration")
```
Comment (thunderboltsid) [inline karbon/karbon_cluster_registration_service_test.go:92]: Can we remove this or switch this to `t.Log` ?

**Comment by thunderboltsid** [comment]
Looks good for most part; please add a entry in CHANGELOG.md

**Review by thunderboltsid — APPROVED** [review]

**Pull request merged**